modules/dnn/src/layers/fully_connected_layer.cpp

```diff
@@ -299,11 +298,14 @@ class FullyConnectedLayerImpl CV_FINAL : public InnerProductLayer
 
             if (use_half)
             {
-                half_blobs.resize(umat_blobs.size());
                 for (int i = 0; i < umat_blobs.size(); i++)
                 {
                     if (!umat_blobs[i].empty())
-                        convertFp16(umat_blobs[i], half_blobs[i]);
+                    {
+                        UMat halfs;
+                        convertFp16(umat_blobs[i], halfs);
+                        umat_blobs[i] = halfs;
```

Comment:
Perhaps we should preserve original blobs due possible network reconfiguration.

Reply:
@alalek, you're right. Fixed that.